src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -117,22 +138,29 @@ public bool IsEmpty
         /// <param name="work">The WorkItem to process</param>
         public void Enqueue(WorkItem work)
         {
-            lock (_syncRoot)
+            do
```

Comment:
I agree with the general statement and I think it fits well with these low overhead operations. @oznetmaster are you using any sort of backing delay before retry? It's hard to fully examine the code on my phone.